nunaliit2-js/src/main/js/nunaliit2/n2.widgetTime.js

```diff
@@ -83,8 +84,504 @@ function formatDate(date, format){
+ * @param {string} containerId - Unique identifier for the container id
+ * @param {string} [sourceModelId] - Unique identifier for the source model id
+ * @param {string} [startDate=null] - Initial start date for the widget
+ * using the date format 'yyyy-mm-dd'
+ * @param {string} [endDate=null] - Initial end date for the widget using
+ * the date format 'yyyy-mm-dd'
+*/
+var DateRangeWidget = $n2.Class({
+
+	dispatchService: null,
+
+	containerId: null,
+
+	sourceModelId: null,
+
+	rangeChangeEventName: null,
+
+	rangeMin: null,
+
+	rangeMax: null,
+
+	intervalChangeEventName: null,
+
+	intervalSetEventName: null,
+
+	intervalMin: null,
+
+	intervalMax: null,
+
+	startDate: null,
+
+	startDatePicker: null,
+
+	endDate: null,
+
+	endDatePicker: null,
+
+	initialize: function(opts_) {
+		var opts = $n2.extend({
+			containerId: null
+			,dispatchService: null
+			,sourceModelId: null
+			,startDate: null
+			,endDate: null
+		},opts_);
+
+		var modelInfoRequest, sourceModelInfo, paramInfo, dispatchFn;
+		var _this = this;
+
+		this.containerId = opts.containerId;
+		this.dispatchService = opts.dispatchService;
+		this.sourceModelId = opts.sourceModelId;
+		this.startDate = opts.startDate;
+		this.endDate = opts.endDate;
+
+		// Set up model listener
+		if (this.dispatchService) {
+			// Get model info
+			modelInfoRequest = {
+				type: 'modelGetInfo'
+				,modelId: this.sourceModelId
+				,modelInfo: null
+			};
+			this.dispatchService.synchronousCall(DH, modelInfoRequest);
+			sourceModelInfo = modelInfoRequest.modelInfo;
+
+			if (sourceModelInfo
+				&& sourceModelInfo.parameters
+				&& sourceModelInfo.parameters.range) {
+				paramInfo = sourceModelInfo.parameters.range;
+				this.rangeChangeEventName = paramInfo.changeEvent;
+
+				if (paramInfo.value) {
+					this.rangeMin = paramInfo.value.min;
+					this.rangeMax = paramInfo.value.max;
+				}
+			}
+
+			if (sourceModelInfo
+				&& sourceModelInfo.parameters
+				&& sourceModelInfo.parameters.interval) {
+				paramInfo = sourceModelInfo.parameters.interval;
+				this.intervalChangeEventName = paramInfo.changeEvent;
+				this.intervalSetEventName = paramInfo.setEvent;
+
+				if (paramInfo.value) {
+					this.intervalMin = paramInfo.value.min;
+					this.intervalMax = paramInfo.value.max;
+				}
+			}
+
+			dispatchFn = function(m, addr, dispatcher) {
+				_this._handle(m, addr, dispatcher);
+			};
+
+			if (this.rangeChangeEventName) {
+				this.dispatchService.register(DH, this.rangeChangeEventName, dispatchFn);
+			}
+
+			if (this.intervalChangeEventName) {
+				this.dispatchService.register(DH, this.intervalChangeEventName, dispatchFn);
+			}
+		}
+
+		if (!this.containerId) {
+			throw new Error('containerId must be specified');
+		}
+
+		this._display();
+	},
+
+	_getWidgetOffset: function() {
+		var $widgetBtn = $('.n2widget_date_range')
+			.find('.n2widget_date_range_button');
+		var offset = $widgetBtn.offset();
+
+		offset.width = $widgetBtn.width();
+		if (offset) {
```

Comment:
Should there be a default?  Nothing will return if it's not set, how do we handle that case?  I suppose there is just no offset in that case?

Reply:
I've added a default offset value. Although it's unlikely to be used, having a default make's the function more defensive, so I agree with the change. So I've included a default offset of 0, 0 for those circumstances when either jQuery stops working, or if the button's class name were to be renamed.